### DISCUSSION TITLE
Why the Endpoint address has the mandatory format: <account-specific-prefix>.iot.<region>.amazonaws.com

### DISCUSSION THREAD

**Discussion by ce-jaro** [aws/aws-iot-device-sdk-java#194] [Q&A] [answered]
It looks like this library requires the endpoint host to be in format of: <account-specific-prefix>.iot.<region>.amazonaws.com.
The MQTT endpoints are usually hidden behind custom domains, like: mqtt-broker-service.companydomain.com.
Why there is hard requirement for endpoint to be in this complex format? why the endpoint address can't be anything which is resolved to some IP-address ? The other MQTT client libraries does not have such us limitation. Is it possible to change this requirment?

**Comment by jmklix** [comment, marked as answer]
This is a design decision that was made IoT core for how to securely connect with AWS's IoT services. This sdk is just following the requirements required to connect to AWS IoT services (This sdk can also be used to connect to other MQTT servers). You can read more about service endpoints [here](https://docs.aws.amazon.com/iot/latest/developerguide/iot-connect-devices.html).

I think you can do what you are looking for by using custom endpoints:
> You can also use your own fully-qualified domain name (FQDN), such as example.com, and the associated server certificate to connect devices to AWS IoT by using [Configurable endpoints](https://docs.aws.amazon.com/iot/latest/developerguide/iot-custom-endpoints-configurable.html).

Does the above custom endpoints work for you? If this needs any changes to this sdk, it is not possible. This sdk is not accepting any more feature requests, but the v2 sdk is (https://github.com/aws/aws-iot-device-sdk-java-v2). Also we recommend using the v2 sdk.
